cypress/integration/requests/patron-initiated/request-item-recall.spec.js

```diff
@@ -0,0 +1,130 @@
+import NewRequest from '../../../support/fragments/requests/newRequest';
+import inventoryInstances from '../../../support/fragments/inventory/inventoryInstances';
+
+describe('ui-requests: Request: Edit requests. Make sure that edits are being saved.', () => {
+  // TODO: A client configured to use edge-patron API
+  const testData = {
+    pickupServicePoint: 'Circ Desk 1',
+    patronGroup: {},
+  };
+  const userData0 = { ...DefaultUser.defaultApiPatron };
+  const userData1 = { ...DefaultUser.defaultApiPatron };
+  userData1.username = `autotest_username_${getRandomPostfix()}`;
+  userData1.barcode = `1234456_${getRandomPostfix()}`;
+  const requestPolicyWithRecall = defaultRequestPolicy;
+  requestPolicyWithRecall.requestTypes.push('Recall', 'Page');
+  const item = {
+    instanceName: `For_request_${getRandomPostfix()}`,
+    itemBarcode: generateItemBarcode(),
+  };
+  const newRequestData = {
+    itemBarcode: item.itemBarcode,
+    itemTitle: testData.instanceName,
+    requesterBarcode: userData1.barcode,
+    pickupServicePoint: testData.pickupServicePoint,
+    userProps: {
+      lastName: userData1.personal.lastName,
+      firstName: userData1.personal.firstName,
+      middleName: userData1.personal.middleName,
+    }
+  };
+
+
+  before('Creating circ rule with request policy, users and item with rolling loan period', () => {
+    cy.getAdminToken();
+
+    // Create request policy that allows recalls and pages
+    RequestPolicy.createApi(requestPolicyWithRecall).then((body) => {
+      testData.requestPolicyId = body.id;
+    });
+
+    // Create 2 users in the same group
+    PatronGroups.createViaApi()
+      .then(res => {
+        testData.patronGroup.id = res;
+        Users.createViaApi({
+          patronGroup: res,
+          ...userData0
+        }).then((createdUser) => {
+          userData0.id = createdUser.id;
+        });
+
+        Users.createViaApi({
+          patronGroup: res,
+          ...userData1
+        }).then((createdUser) => {
+          userData1.id = createdUser.id;
+        });
+      });
+
+    // Create item with remembered instance type
+    inventoryInstances.createInstanceViaApi(item.instanceName, item.itemBarcode);
+    ServicePoints.getViaApi({ limit: 1, query: 'pickupLocation=="true"' })
+      .then((servicePoints) => {
+        UserEdit.addServicePointViaApi(servicePoints[0].id, userData0.id).then((points) => {
+          testData.userServicePoint = points.body.defaultServicePointId;
+        });
+      });
+
+
+    // Create a circulation rule with an associated request policy that allows recalls and pages
+    cy.getCirculationRules().then((res) => {
+      testData.baseRules = res.rulesAsText;
+      testData.ruleProps = CirculationRules.getRuleProps(res.rulesAsText);
+      testData.ruleProps.r = testData.requestPolicyId;
+
+      CirculationRules.addRuleApi(res.rulesAsText, testData.ruleProps, 'g ', testData.patronGroup.id).then(() => {
```

Comment:
I have not founded the precondition of the test case related to the special loan police:
Item(s) that are checked out using a loan policy that uses rolling loan periods

Can you add please a loan policy that uses rolling loan periods?